src/lib/dai.c

```diff
@@ -66,8 +66,7 @@ struct dai *dai_get(uint32_t type, uint32_t index, uint32_t flags)
 		if (!ret)
 			d->sref++;
 
-		trace_dai("dai_get(), d = %p, sref = %d",
-			  (uintptr_t)d, d->sref);
+		trace_dai("dai_get(), d = %p, sref = %d", d, d->sref);
```

Comment:
You still need cast here

Reply:
@jajanusz That removal did look funny to me too. The trace functions are weird anyway because they only support integer parameters even if %p is accepted. Could one add some documentation stating this weird behavior to remove further instances of such confusion?

Reply:
@paulstelian97 Imo these trace functions overused metaprogramming, it's unreadable mess, but yea, we will have to do something with that...

Reply:
Why do we need a cast for `%p`?

Reply:
@finikorg Internally the pointer is stored into an "int" variable due to the weird implementation of the trace functions. So the cast is indeed required.

(note: etrace should be enabled from config; the default config normally does enable it).

As I said earlier, this is confusing and the trace functions require some extra documentation stating this quirk (that all parameters must be of integral types even with the %p and related specifiers).

Another quirk is that the %s specifier will probably not work too well (it may work for constant strings but dynamically generated ones are out of the question; the cast to integer is also needed)